crash-handler-process/platforms/upload-window-win.cpp

```diff
@@ -0,0 +1,413 @@
+		SetWindowText(upload_label_hwnd, upload_progress_message);
+		EnableWindow(ok_button_hwnd, true);
+		EnableWindow(cancel_button_hwnd, false);
+		break;
+	}
+	case CUSTOM_UPLOAD_CANCELED: {
+		swprintf(upload_progress_message, L"Upload cancleled. You may share the file with our support manually.");
+		SetWindowText(upload_label_hwnd, upload_progress_message);
+		EnableWindow(ok_button_hwnd, true);
+		EnableWindow(cancel_button_hwnd, false);
+		break;
+	}
+	case CUSTOM_UPLOAD_FAILED: {
+		SetWindowText(upload_label_hwnd, L"Uploading failed. You may share the file with our support manually.");
+		EnableWindow(ok_button_hwnd, true);
+		EnableWindow(cancel_button_hwnd, false);
+		break;
+	}
+	case WM_COMMAND: {
+		if ((HWND)lParam == ok_button_hwnd) {
+			EnableWindow(ok_button_hwnd, false);
+			EnableWindow(cancel_button_hwnd, false);
+			SetWindowText(upload_label_hwnd, L"");
+			button_clicked = IDOK;
+			upload_window_choose_variable.notify_one();
+			break;
+		}
+		if ((HWND)lParam == cancel_button_hwnd) {
+			EnableWindow(ok_button_hwnd, false);
+			EnableWindow(cancel_button_hwnd, false);
+			SetWindowText(upload_label_hwnd, L"");
+			button_clicked = IDCANCEL;
+			upload_window_choose_variable.notify_one();
+			break;
+		}
+		break;
+	}
+	case CUSTOM_REQUESTED_CLICK: {
+		if (!upload_window_hwnd) {
+			button_clicked = IDOK;
+			upload_window_choose_variable.notify_one();
+		}
+		break;
+	}
+	}
+
+	return DefWindowProc(hwnd, msg, wParam, lParam);
+}
+
+UploadWindow::UploadWindow() {}
+
+UploadWindow::~UploadWindow()
+{
+	if (window_thread) {
+		PostMessage(upload_window_hwnd, CUSTOM_CLOSE_MSG, 0, 0);
+		if (window_thread->joinable())
+			window_thread->join();
+	}
+};
+
+UploadWindow* UploadWindow::getInstance()
+{
+	if (instance)
+		return instance;
+	instance = new UploadWindow();
+	return instance;
+}
+
+void UploadWindow::shutdownInstance()
+{
+	delete instance;
+	instance = nullptr;
+}
+
+void UploadWindow::windowThread()
+{
+	log_info << "UploadWindow windowThread started" << std::endl;
+	hInstance = GetModuleHandle(NULL);
+
+	WNDCLASSEX wc;
+	wc.cbSize        = sizeof(WNDCLASSEX);
+	wc.style         = CS_NOCLOSE;
+	wc.lpfnWndProc   = FrameWndProc;
+	wc.cbClsExtra    = 0;
+	wc.cbWndExtra    = 0;
+	wc.hInstance     = hInstance;
+	wc.hCursor       = LoadCursor(NULL, IDC_ARROW);
+	wc.hbrBackground = CreateSolidBrush(RGB(23, 36, 45));
+	wc.lpszMenuName  = NULL;
+	wc.lpszClassName = TEXT("uploaderwindow");
+
+	if (!RegisterClassEx(&wc)) {
+		log_error << "Failed to create a class for uploader window " << GetLastError() << std::endl;
+		upload_window_choose_variable.notify_one();
+		return;
+	}
+
+	/* We only care about the main display */
+	int screen_width  = GetSystemMetrics(SM_CXSCREEN);
+	int screen_height = GetSystemMetrics(SM_CYSCREEN);
+
+	upload_window_hwnd = CreateWindowEx(
+	    WS_EX_CLIENTEDGE,
+	    TEXT("uploaderwindow"),
+	    TEXT("Streamlabs OBS Crash Dump Uploader"),

```

Comment:
Send additional debug information to the developers.

Might be a better message